src/Hosting/Hosting/test/HostTests.cs

```diff
@@ -81,6 +82,56 @@ public void CreateDefaultBuilder_EnablesValidateOnBuild()
             Assert.Throws<AggregateException>(() => hostBuilder.Build());
         }
 
+        [Fact]
+        public void CreateDefaultBuilder_ConfigJsonDoesNotReload()
+        {
+            Environment.SetEnvironmentVariable("DOTNET_HOSTBUILDER_CONFIG_RELOAD", "false");
+            Environment.CurrentDirectory = Path.GetTempPath();
+
+            string SaveRandomConfig()
+            {
+                var newMessage = $"Hello ASP.NET Core: {Guid.NewGuid():N}";
+                File.WriteAllText(Path.Combine(Path.GetTempPath(), "appsettings.json"), $"{{ \"Hello\": \"{newMessage}\" }}");
+                return newMessage;
+            }
+
+            var dynamicConfigMessage1 = SaveRandomConfig();
+            var host = Host.CreateDefaultBuilder().Build();
+            var config = host.Services.GetRequiredService<IConfiguration>();
+
+            Assert.Equal(dynamicConfigMessage1, config["Hello"]);
+
+            var dynamicConfigMessage2 = SaveRandomConfig();
+            Task.Delay(1000).Wait(); // Give reload time to fire if it's going to.
```

Comment:
This is going to be unreliable. You can directly register for the change callback.
https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.iconfiguration.getreloadtoken?view=dotnet-plat-ext-3.1#Microsoft_Extensions_Configuration_IConfiguration_GetReloadToken

Reply:
This is where we're going to run into a hard problem, the feature is testing that we've disabled (default builder's) filesystem modification reloads, so even if I subscribe to the notification being raised, that's already downhill from the filesystem notification needing to have triggered, or not, and I'm still going to have to wait X time for that event right?

I can assert that the event fired, but since the backend is asynchronous, in the instance that we're trying to assert that this did happen, it's not only the same, but I need to wait anyway, otherwise what I've done is test immediately before the change had time to propagate, and in the case we're trying to assert that the filesystem event did not fire, then I still need to have waited a reasonable but non-making-the-tests-unbearable amount of time for it to have done so.

Reply:
Fair point. The test for disabled reload can wait one second and ensure check that config did not change. Just make sure to do so properly: (.Wait() can cause threading issues even in the tests.)
```suggestion
            await Task.Delay(1000); // Give reload time to fire if it's going to.
```

For the positive test we must use the event and a long timeout. The test machines have strange load characteristics and are subject to 1-10s random delay.

Reply:
Okay cool, that makes sense. I tried to hybridize this approach you mentioned. I increased the timeout to be 10 seconds since as you mentioned this can be volatile, but used the ReloadToken to cancel the Task if the ReloadTokoen fires, since we know that means we're ready to proceed regardless.